### ISSUE TITLE
Enhancement Request: Next Action view

### ISSUE THREAD

**Issue by richard-edwards** [budowski/budoist#2]
Here is something I think would be a really great addition to this type of app. In any view, project, labels, queries if you could go into a 'Next Action' mode it would only show the next action on the list, 1 at a time in a bigger font centered in the screen with a 'Complete' or 'Done' button or checkbox. After you have checked off that action, it would show the next in the list etc. etc.

This would really help people focus on the task at hand. If the view is synced in the background and the currently shown action isn't now the top in the list it should start back at the top of the list again. I was doing something like this with the Todoist API and a desktop app but it took up too much screen real estate and kept getting hidden.

**Comment by budowski** [comment]
Good idea, but how should Budoist determine what the next action should be?
On Mar 23, 2012 3:39 AM, "redwards1966" <
reply@reply.github.com>
wrote:

**Comment by richard-edwards** [comment]
I would just do them in whatever order they are in the list. For instance
if I run the query 'Overdue, upcoming and important' it shows me a list in
a specific order. I would just stay with that order and take the first one
in the list for the first 'next action', once that is done go to the next.
When nothing is in the list then say something like 'All done! Take a
break' :)

The user would have to be in either a project, label or query for this to
work properly. It doesn't make send to go into it off the project list.

For instance, say I had a label called 'Calls' or calls to make. I would
hit labels, choose 'Calls' which takes me to a list of calls. Once there I
would hit an icon that would put it into 'Next Action' mode. Perhaps it
would just be '1 Up Mode' or 'Single Mode' instead and also have back and
forth buttons to loop through the current list instead of only showing the
next action. It might be more flexible for people that way, especially if
the next action is something they couldn't complete (line was busy on a
phone call) and they wanted to jump to the next thing on the list.

On Fri, Mar 23, 2012 at 10:13 AM, budowski <
reply@reply.github.com

> wrote:
> 
> Good idea, but how should Budoist determine what the next action should be?
> On Mar 23, 2012 3:39 AM, "redwards1966" <
> reply@reply.github.com>
> wrote:
> 
> > Here is something I think would be a really great addition to this type
> > of
> > app. In any view, project, labels, queries if you could go into a 'Next
> > Action' mode it would only show the next action on the list, 1 at a time
> > in
> > a bigger font centered in the screen with a 'Complete' or 'Done' button
> > or
> > checkbox. After you have checked off that action, it would show the next
> > in
> > the list etc. etc.
> > 
> > This would really help people focus on the task at hand. If the view is
> > synced in the background and the currently shown action isn't now the top
> > in the list it should start back at the top of the list again. I was
> > doing
> > something like this with the Todoist API and a desktop app but it took up
> > too much screen real estate and kept getting hidden.
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/budowski/budoist/issues/2
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/budowski/budoist/issues/2#issuecomment-4659664

## 

Regards,

Richard
